### PR TITLE
Add bot to repo that closes stale issues

### LINKED ISSUE #7344: Add bot to repo that closes stale issues
Lots of other MS teams have this set up already, check with them to see what is required.

### PR DESCRIPTION
**Pull request by StellaHuang95**
Add a bot to close the 'waiting for response' issues after 30 days inactivity.

fixes #7344

### CODE REVIEW

File: .github/workflows/stale.yml
Comment (AdamYoblick) [inline .github/workflows/stale.yml:25]: Does this mean that `waiting for response` is the only label that can be on the issue, or does it mean that it will close it as long as it has `waiting for response` as a label?

I would prefer the latter if possible.

File: .github/workflows/stale.yml
Comment (AdamYoblick) [inline .github/workflows/stale.yml:24]: I would actually prefer a shorter amount of time here, as long as Jude agrees. I think two weeks is more than enough time to wait, but if he wants longer that's fine 😄

**Review by AdamYoblick — APPROVED** [review]
Approved with comments/questions

File: .github/workflows/stale.yml
Comment (StellaHuang95) [reply .github/workflows/stale.yml]: @AdamYoblick , thanks, I've changed it to `any-of-labels`

File: .github/workflows/stale.yml
Comment (StellaHuang95) [reply .github/workflows/stale.yml:24]: ok I will ask him and update later if needed.

**Comment by AdamYoblick** [comment]
@StellaHuang95 After this is merged in, can you please create a dummy issue, mark is as Waiting for response, and see if the bot does what it's supposed to do after 30 days? 😄

**Pull request merged**
